packages/shared/__tests__/describeComponentFrame-test.js

```diff
@@ -88,15 +90,21 @@ describe('Component stack trace displaying', () => {
       'C:\\funny long (path)/index.js': 'funny long (path)/index.js',
       'C:\\funny long (path)/index.jsx': 'funny long (path)/index.jsx',
     };
-    Object.keys(fileNames).forEach((fileName, i) => {
+
+    const root = ReactDOMClient.createRoot(container);
+
+    let i = 0;
+    for (const fileName in fileNames) {
```

Comment:
```
for (let i = 0; i < fileNames.length; i++)

```

or 

```
for (let [i, fileName] of fileNames.entries()) {
```

?

Reply:
`fileNames` is an object so a basic `for` loop would also be awkward. We already had precedent below to iterate over `fileNames` with `for-in` and manual increments. 

`for-of` would trigger lint violations and while it's likely a false-positive for tests, I just stuck with what's already there. But I agree that `for-in` with increments is pretty unorthodox.